jwst/outlier_detection/outlier_detection.py

```diff
@@ -405,49 +405,49 @@ def flag_cr(sci_image, blot_image, **pars):
     sci_data = sci_image.data
     blot_data = blot_image.data
     blot_deriv = abs_deriv(blot_data)
-
     err_data = np.nan_to_num(sci_image.err)
 
-    # Define output cosmic ray mask to populate
-    cr_mask = np.zeros(sci_image.shape, dtype=np.uint8)
-
     # create the outlier mask
-    if pars.get('resample_data', True):  # dithered outlier detection
+    if resample_data:  # dithered outlier detection
         blot_data += subtracted_background
         diff_noise = np.abs(sci_data - blot_data)
 
-        # create an initial mask based on
-        # a scaled version of the derivative image (dealing with interpolating issues?)
+        # Create a boolean mask based on a scaled version of
+        # the derivative image (dealing with interpolating issues?)
         # and the standard n*sigma above the noise
-        t2 = scl1 * blot_deriv + snr1 * err_data
-        # in this mask
-        init_mask = np.logical_not(np.greater(diff_noise, t2))
+        threshold1 = scale1 * blot_deriv + snr1 * err_data
+        mask1 = np.greater(diff_noise, threshold1)
 
-        # Convolve mask with 3x3 kernel
-        kernel = np.ones((3, 3), dtype=np.uint8)
-        init_mask_smoothed = np.zeros(init_mask.shape, dtype=np.int32)
-        ndimage.convolve(init_mask, kernel, output=init_mask_smoothed, mode='nearest', cval=0)
+        # Smooth the boolean mask with a 3x3 boxcar kernel
+        kernel = np.ones((3, 3), dtype=int)
+        mask1_smoothed = ndimage.convolve(mask1, kernel, mode='nearest')
 
-        # create a 2nd mask like the first, but based on the 2nd set of
+        # Create a 2nd boolean mask based on the 2nd set of
         # scale and threshold values
-        mask_2ndpass = scl2 * blot_deriv + snr2 * err_data
+        threshold2 = scale2 * blot_deriv + snr2 * err_data
+        mask2 = np.greater(diff_noise, threshold2)
 
-        cr_mask = np.logical_not(np.greater(diff_noise, mask_2ndpass) & np.less(init_mask_smoothed, 9))
+        # Final boolean mask
+        cr_mask = mask1_smoothed & mask2
 
     else:  # stack outlier detection
         diff_noise = np.abs(sci_data - blot_data)
 
-        # straightforward detection of outliers for non-dithered data
-        #   as err_data includes all noise sources (photon, read, and flat for baseline)
-        cr_mask = np.logical_not(np.greater(diff_noise, snr1 * err_data))
+        # straightforward detection of outliers for non-dithered data since
+        # err_data includes all noise sources (photon, read, and flat for baseline)
+        cr_mask = np.greater(diff_noise, snr1 * err_data)
```

Comment:
Shouldn't `logical_not(greater(...))` be `less_equal(...)`?

Reply:
I removed the `np.logical_not` here and removed the `np.invert` below where it is actually used to create the bitmask.

Reply:
Oh, `invert(logical_not(greater))`...